### PR TITLE
Create deployments role

### PR DESCRIPTION
**Pull request by nickromneydfe**
## Jira ticket URL

https://dfedigital.atlassian.net/browse/TEVA-2161

## Changes in this PR:

- create deployers group
- allow members of deployers group to assume deployments role, without needing MFA
- attach existing policies associated with the deploy user to the deployments role
- support role and policy attachment in `common/modules/domains` 

## Testing

Add `deploy` user to Deployers group
With the keys for `deploy` user, assume the role
```
aws sts assume-role --role-arn arn:aws:iam::530003481352:role/Deployments --role-session-name deploy
```

## Future PR

- Use assumed role in CI/CD
- Remove direct policy attachments to the `deploy` user

### CODE REVIEW

File: terraform/common/groups.tf [hunk 1]
```diff
@@ -269,12 +286,46 @@ resource "aws_iam_policy" "parameter_store" {
...
 resource "aws_iam_role_policy_attachment" "secreteditor_role_policies" {
   role       = aws_iam_role.secreteditor.name
   policy_arn = aws_iam_policy.parameter_store.arn
 }
 
+# Deployers group, deployments role - no MFA required
+
+resource "aws_iam_role" "deployments" {
+  name               = "Deployments"
+  assume_role_policy = data.aws_iam_policy_document.assume_role_no_mfa.json
```
Comment (saliceti) [inline terraform/common/groups.tf:298]: This should be the policy not the document

**Comment by saliceti** [comment]
As discussed please add a policy for the deploy user to assumer the role deployments

File: terraform/common/groups.tf (hunk 1 above)
Comment (nickromneydfe) [reply terraform/common/groups.tf:298]: The other four references to `assume_role_policy` all directly consume the json, e.g.

```
resource "aws_iam_role" "secreteditor" {
  name               = "SecretEditor"
  assume_role_policy = data.aws_iam_policy_document.assume_role_if_mfa_present.json
}
```

Comment (saliceti) [reply terraform/common/groups.tf:298]: ok 👍

**Comment by nickromneydfe** [comment]
> As discussed please add a policy for the deploy user to assumer the role deployments

Not needed if the deploy user is granted membership of the Deploymers group which can assume the role Deployments

**Review by saliceti — APPROVED** [review]

**Pull request merged**